src/shogun/features/RandomKitchenSinksDotFeatures.cpp

```diff
@@ -37,14 +38,13 @@ SGMatrix<float64_t> RandomKitchenSinksDotFeatures::generate_random_coefficients(
 {
 	SGVector<float64_t> vec = generate_random_parameter_vector();
 	SGMatrix<float64_t> random_params(vec.vlen, num_samples);
-	for (index_t dim=0; dim<random_params.num_rows; dim++)
-		random_params(dim, 0) = vec[dim];
+	linalg::range_fill(random_params,0.0);
```

Comment:
why `range_fill`?

Reply:
I'm initiating the values with zero.

Reply:
but `range_fill` is it initialise values from `[0, 1,..., size]`, no? there's a function for zeroing out

Reply:
You're right. ```random_params.zero()``` should do the trick !